### PR TITLE
Remotesensing

### PR DESCRIPTION
**Pull request by CharlesAuthier**

### CODE REVIEW

File: deep-learning/_posts/2018-01-16-3D-FCNN.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+author: Charles Authier
+pdf:    www.mdpi.com/2072-4292/9/11/1139/pdf   
+cite:
+  authors: "Shaohui Mei, Xin Yuan, Jingyu Ji, Yifan Zhang, Shuai Wan and Qian Du"
+  title:   "Hyperspectral Image Spatial Super-Resolution via 3D Full Convolutional Neural Network"
+  venue:   "Remote Sens. 2017"
+---
+
+### Description
+In this paper, they want to increase the spatial resolution for hyperspectral images to match with the deferent spacial resolution of the different captors. For that they created a three-dimensional full CNN (3D-FCNN) is constructed for spatial Super Resolution of hyperspectral images.
```
Comment (Dref360) [inline deep-learning/_posts/2018-01-16-3D-FCNN.md]: different

File: deep-learning/_posts/2018-01-16-3D-FCNN.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+To do that they exploit both the spatial context of neighboring pixels and spectral correlation of neighboring bands. Changing the 2D covultion to a 3D convolution.
```
Comment (Dref360) [inline deep-learning/_posts/2018-01-16-3D-FCNN.md]: convolution

File: deep-learning/_posts/2018-01-16-3D-FCNN.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+<img src="/deep-learning/images/3DFCNN/2d_math.png" width="350"><img src="/deep-learning/images/3DFCNN/3d_math.png" width="350">
+
+<img src="/deep-learning/images/3DFCNN/2D.png" width="325"><img src="/deep-learning/images/3DFCNN/3D.png" width="375">
+
+The architecture of the proposed 3D-FCNN fully explores spatial-spectral features in the data cube. First, because of the low resolution of the input, they up-scaled the input of the desired output size using a bicubic interpolation. At the same time, the up-scaling alleviating the spectral distortion during the convolution. In the structure, we find four convolutional layers and the input image will be $$ 33 \times 33 \times c \times 1$$, where $$33 \times 33$$ is the spatial dimension, $$c$$ the number of spectral layers and the $$ 1$$ is the color channel for the HSI (Hyperspectral imaging). Between those four, the 'ReLU' function is used as activation function for nonlinear mapping.
```
Comment (Dref360) [inline deep-learning/_posts/2018-01-16-3D-FCNN.md]: upscale

File: deep-learning/_posts/2018-01-16-3D-FCNN.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+<img src="/deep-learning/images/3DFCNN/framwork_3d.png" width="500">
+
+In the table below, we have the size of the different kernel associated with layers.
+
+<img src="/deep-learning/images/3DFCNN/table_archi.png" width="700">
+
+To provide border effects they don't add any padding in any convolutions. At the end, is the MSE (mean squared error) between outputs and ground-truth who is used as a cost function.
+
+Well, train with good parameters they want to use this trained network on different datasets obtained from the same sensor.
```
Comment (Dref360) [inline deep-learning/_posts/2018-01-16-3D-FCNN.md]: To rewrtie

File: deep-learning/_posts/2018-01-16-3D-FCNN.md [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+---
+
+### Description
+In this paper, they want to increase the spatial resolution for hyperspectral images to match with the deferent spacial resolution of the different captors. For that they created a three-dimensional full CNN (3D-FCNN) is constructed for spatial Super Resolution of hyperspectral images.
+
+To do that they exploit both the spatial context of neighboring pixels and spectral correlation of neighboring bands. Changing the 2D covultion to a 3D convolution.
+
+<img src="/deep-learning/images/3DFCNN/2d_math.png" width="350"><img src="/deep-learning/images/3DFCNN/3d_math.png" width="350">
+
+<img src="/deep-learning/images/3DFCNN/2D.png" width="325"><img src="/deep-learning/images/3DFCNN/3D.png" width="375">
```
Comment (Dref360) [inline deep-learning/_posts/2018-01-16-3D-FCNN.md]: fig2 is useless imo

File: deep-learning/_posts/2018-01-16-3D-FCNN.md [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+### Results
+In this paper, they use the mean peak signal-to-noise ratio (MPSNR) and the mean structural similarity (MSSIM) to evaluate the spatial performance of the system. To provide a score on the spectral performance is the spectral angle mapper (SAM) between the reconstructed spectra and their corresponding ground-truth spectra who is used. To simulate low spatial-resolution HSIs, they down-sampled the images by a factor of 2 and to test the robustness of the architecture they add different levels of noise (SNR or signal-to-noise ratio).
+
+<img src="/deep-learning/images/3DFCNN/result_table.png" width="700">
+
+<img src="/deep-learning/images/3DFCNN/resultat_images2.png" width="700">
+
+The results show a clear victory for the 3D-FCNN even with noise. The proposed 3D-FCNN based algorithm also achieves best spatial reconstruction for all the cases. The proposed 3D-FCNN based algorithm provides much better spectral fidelity no matter the dataset.
+
+But those results have a cost, on an NVIDIA GTX1070 with 16GB memory, that takes 20h to train the model.
```
Comment (Dref360) [inline deep-learning/_posts/2018-01-16-3D-FCNN.md]: For how many samples?

File: deep-learning/_posts/2018-01-16-3D-FCNN.md (hunk 5 above)
Comment (Dref360) [reply deep-learning/_posts/2018-01-16-3D-FCNN.md]: Fig 3 as well

File: deep-learning/_posts/2018-01-16-3D-FCNN.md [hunk 7]
```diff
@@ -0,0 +1,45 @@
+---
+layout: review
+title:  "Hyperspectral Image Spatial Super-Resolution via 3D Full Convolutional Neural Network"
+tags:   hyperspectral
```
Comment (jodoin) [inline deep-learning/_posts/2018-01-16-3D-FCNN.md]: Add more meaningful tags.  See table https://vitalab.github.io/howto/.  Also, add "hyperspectral" to the table.

File: deep-learning/_posts/2018-01-16-3D-FCNN.md [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+author: Charles Authier
+pdf:    www.mdpi.com/2072-4292/9/11/1139/pdf   
+cite:
+  authors: "Shaohui Mei, Xin Yuan, Jingyu Ji, Yifan Zhang, Shuai Wan and Qian Du"
+  title:   "Hyperspectral Image Spatial Super-Resolution via 3D Full Convolutional Neural Network"
+  venue:   "Remote Sens. 2017"
+---
+
+### Description
+In this paper, they want to increase the spatial resolution for hyperspectral images to match with the different spacial resolution of the different captors. For that they created a three-dimensional full CNN (3D-FCNN) is constructed for spatial Super Resolution of hyperspectral images.
```
Comment (jodoin) [inline deep-learning/_posts/2018-01-16-3D-FCNN.md]: of hyperspectral

Comment (jodoin) [reply deep-learning/_posts/2018-01-16-3D-FCNN.md]: (3D-FCNN) to perform super resolution of...

File: Gemfile.lock [hunk 9]
```diff
@@ -51,7 +51,7 @@ DEPENDENCIES
   tzinfo-data
 
 RUBY VERSION
-   ruby 2.4.0p0
+   ruby 2.3.1p112
```
Comment (lemairecarl) [inline Gemfile.lock]: Strange. Seems like you don't have the same version of Ruby as us.

File: deep-learning/_posts/2018-01-16-3D-FCNN.md [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+author: Charles Authier
+pdf:    www.mdpi.com/2072-4292/9/11/1139/pdf   
+cite:
+  authors: "Shaohui Mei, Xin Yuan, Jingyu Ji, Yifan Zhang, Shuai Wan and Qian Du"
+  title:   "Hyperspectral Image Spatial Super-Resolution via 3D Full Convolutional Neural Network"
+  venue:   "Remote Sens. 2017"
+---
+
+### Description
+In this paper, they want to increase the spatial resolution of hyperspectral images to match with the different spacial resolution of the different captors. For that they created a three-dimensional full CNN (3D-FCNN) to perform super resolution of hyperspectral images.
```
Comment (lemairecarl) [inline deep-learning/_posts/2018-01-16-3D-FCNN.md]: spatial
captor = sensor?

File: deep-learning/_posts/2018-01-16-3D-FCNN.md [hunk 11]
```diff
@@ -0,0 +1,45 @@
...
+
+To do that they exploit both the spatial context of neighboring pixels and spectral correlation of neighboring bands. Changing the 2D convoltion to a 3D convolution.
```
Comment (lemairecarl) [inline deep-learning/_posts/2018-01-16-3D-FCNN.md]: convolution

File: deep-learning/_posts/2018-01-16-3D-FCNN.md [hunk 12]
```diff
@@ -0,0 +1,45 @@
...
+
+<img src="/deep-learning/images/3DFCNN/2d_math.png" width="350">
+
+<img src="/deep-learning/images/3DFCNN/2D.png" width="325"><img src="/deep-learning/images/3DFCNN/3D.png" width="375">
+
+The architecture of the proposed 3D-FCNN fully explores spatial-spectral features in the data cube. First, because of the low resolution of the input, they upscale the input of the desired output size using a bicubic interpolation. At the same time, the upscale alleviating the spectral distortion during the convolution. In the structure, we find four convolutional layers and the input image will be $$ 33 \times 33 \times c \times 1$$, where $$33 \times 33$$ is the spatial dimension, $$c$$ the number of spectral layers and the $$ 1$$ is the color channel for the HSI (Hyperspectral imaging). Between those four, the 'ReLU' function is used as activation function for nonlinear mapping.
```
Comment (lemairecarl) [inline deep-learning/_posts/2018-01-16-3D-FCNN.md]: >At the same time, the upscale alleviating the spectral distortion during the convolution.

There's something wrong with this sentence.

File: deep-learning/_posts/2018-01-16-3D-FCNN.md [hunk 13]
```diff
@@ -0,0 +1,45 @@
...
+
+<img src="/deep-learning/images/3DFCNN/framwork_3d.png" width="500">
+
+In the table below, we have the size of the different kernel associated with layers.
+
+<img src="/deep-learning/images/3DFCNN/table_archi.png" width="700">
+
+To provide border effects they don't add any padding in any convolutions. At the end, is the MSE (mean squared error) between outputs and ground-truth who is used as a cost function.
```
Comment (lemairecarl) [inline deep-learning/_posts/2018-01-16-3D-FCNN.md]: it is the MSE
that is used

File: deep-learning/_posts/2018-01-16-3D-FCNN.md [hunk 14]
```diff
@@ -0,0 +1,45 @@
...
+
+Train with good parameters they want to use this trained network on different datasets obtained from the same sensor.
+
+### Results
+In this paper, they use the mean peak signal-to-noise ratio (MPSNR) and the mean structural similarity (MSSIM) to evaluate the spatial performance of the system. To provide a score on the spectral performance is the spectral angle mapper (SAM) between the reconstructed spectra and their corresponding ground-truth spectra who is used. To simulate low spatial-resolution HSIs, they down-sampled the images by a factor of 2 and to test the robustness of the architecture they add different levels of noise (SNR or signal-to-noise ratio).
```
Comment (lemairecarl) [inline deep-learning/_posts/2018-01-16-3D-FCNN.md]: use 'who' only when talking about people

File: deep-learning/_posts/2018-01-16-3D-FCNN.md [hunk 15]
```diff
@@ -11,28 +11,28 @@ cite:
...
+The architecture of the proposed 3D-FCNN fully explores spatial-spectral features in the data cube. First, because of the low resolution of the input, they upscale the input of the desired output size using a bicubic interpolation. This step alleviating the spectral distortion during the 3D convolution. In the structure, we find four convolutional layers and the input image will be $$ 33 \times 33 \times c \times 1$$, where $$33 \times 33$$ is the spatial dimension, $$c$$ the number of spectral layers and the $$ 1$$ is the color channel for the HSI (Hyperspectral imaging). Between those four, the 'ReLU' function is used as activation function for nonlinear mapping.
```
Comment (lemairecarl) [inline deep-learning/_posts/2018-01-16-3D-FCNN.md]: alleviates

**Pull request merged**